src/modules/billing/services/transactions-service.js

```diff
@@ -91,6 +91,19 @@ const getBatchTransactions = batch => flatMap(batch.invoices.map(getInvoiceTrans
 
 const getTransactionId = transaction => transaction.id;
 
+const getBatchTransactionHistory = async batchId => {
+  const historicTransactions = await newRepos.billingTransactions.findHistoryByBatchId(batchId);
+
+  // get licences that need to have the 2nd part recalculated for supplementary
+  const twoPartTariffChargeVersionYears = await newRepos.billingBatchChargeVersionYears.findTwoPartTariffByBatchId(batchId);
+
+  // filter the transaction so that it does not include any 2PT transactions where there is no 2PT charge version year
+  return historicTransactions.filter(trx =>
```

Comment:
This works, so it's "fine" but does look a bit messy. To get rid of this **in the future** I would like us to slot a flag in the DB that tells us whether a given transaction is a first part, a second part or neither.